exercises/nth-prime/example.lua

```diff
@@ -0,0 +1,18 @@
+return function(n)
+  assert(n > 0, 'n must be positive')
+
+  local primes = {}
+
+  local function is_prime(n)
+    for _, p in ipairs(primes) do
+      if n % p == 0 then return false end
+      if p^2 > n then return true end
```

Comment:
Interestingly, this little optimization takes the run time of the tests from from ~3.5 seconds to ~0.1 seconds on my machine